### PR TITLE
Retrofit+mockserver

### PR DESCRIPTION
**Pull request by radaladar**
implemented retrofit & mock server, restructured project to be more MVC
most likely to cause conflicts with main branch due to restructuring, removing volley dependency and recreating User class without lombok

### CODE REVIEW

File: app/src/main/java/com/greenfox/peridot/peridot_coz_android/model/api/MockService.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import retrofit2.Call;
+
+public class MockService implements UserInterface {
+
+    public MockService() {
+    }
+
+    @Override
+    public Call<User> getUser() {
+        return null;
```
Comment (adamgyulavari) [inline app/src/main/java/com/greenfox/peridot/peridot_coz_android/model/api/MockService.java]: @radaladar why not return a User here?

File: app/src/main/java/com/greenfox/peridot/peridot_coz_android/activities/MainActivity.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+
+import com.greenfox.peridot.peridot_coz_android.R;
+
+public class MainActivity extends AppCompatActivity {
+
+    // User primary key used by Retrofit - TO BE REVIEWED
+    private final static String USER_ID = "";
```
Comment (adamgyulavari) [inline app/src/main/java/com/greenfox/peridot/peridot_coz_android/activities/MainActivity.java]: @radaladar you dont actually need this here at this point

File: app/src/main/java/com/greenfox/peridot/peridot_coz_android/model/api/MockService.java (hunk 1 above)
Comment (radaladar) [reply app/src/main/java/com/greenfox/peridot/peridot_coz_android/model/api/MockService.java]: we thought retrofit 2.0 would only work if we packed User into a Call, as per the Dev's video ( https://www.youtube.com/watch?v=t34AQlblSeE ), now I realize its not necessary, will be fixed in next commit

File: app/src/main/java/com/greenfox/peridot/peridot_coz_android/activities/MainActivity.java (hunk 2 above)
Comment (radaladar) [reply app/src/main/java/com/greenfox/peridot/peridot_coz_android/activities/MainActivity.java]: fixed in next commit

**Comment by radaladar** [comment]
@adamgyulavari fixed problem with files, circleCI checks pass

**Pull request merged**